Simplify the go quickstart

chalin opened the pull request:
Fixes #213

chalin commented on content/docs/quickstart/go.md:
````diff
@@ -3,122 +3,54 @@ title: Go Quick Start
 
- 5. Setup the example as a module:
+ 1. [Dowload the repo as a zip file][download] and unzip it, or clone
+    the repo:
 
     ```sh
-    $ go mod init examples/helloworld
+    $ git clone https://github.com/grpc/grpc-go
````
@dfawley - in the end, it turns out to be easier to just clone the repo to get the sources. Having the example as a separate module still helps though!

dfawley commented on content/docs/quickstart/go.md:
````diff
@@ -3,122 +3,54 @@ title: Go Quick Start
-     3. Update your environment's path variable to include the path to the
-        `protoc` executable. For example:
+      ```sh
+      $ go get github.com/golang/protobuf/protoc-gen-go@v1.3.5
````
Is it a good idea to specify a patch release here?  If there are bug fixes, a later version of the same minor release would be preferred.  But maybe we shouldn't specify any version?  Future minor versions should always be backward compatible.

... And, they've now deprecated this package, so we'll have to update these instructions to use the v2 implementation once we're ready -- I think we're blocked on https://github.com/grpc/grpc-go/pull/3453 (me) for this.

chalin replied:
> ... so we'll have to update these instructions to use the v2 implementation once we're ready ...

Right. I see this PR as another iterative improvement of the go quick start. Once we're in a position to use the new plugin, I'll update the instructions and drop the `@v1.3.5` version specification.

dfawley replied:
How about drop the `.5` for now, anyway, so we don't pin to a specific version?  I doubt they'll ever release 1.3.6 now that they're on 1.4.x, but if they do it will be for a very good reason.

chalin replied:
Done.

chalin commented:
@dfawley - ping. Is this incremental improvement of the Go quick start good to go?

chalin commented:
@dfawley - I've done the change you suggested (and added `export GO111MODULE=on`  just before the command in case module mode isn't enabled). PTAL

chalin commented:
@dfawley - anything left to do before I can merge?

chalin commented:
@dfawley - I've addressed your concerns, and you likely have your hands full atm, so I'll go ahead and merge. We can address any further issues in another PR.

dfawley commented:
Sorry - yes this was good, thanks.

chalin commented:
Thank you for confirming! 👍
